awx/main/tasks/callback.py

```diff
@@ -0,0 +1,188 @@
+# python
+import json
+import time
+import logging
+
+# Django
+from django.utils.timezone import now
+
+
+# AWX
+from awx.main.tasks.jobs import RunInventoryUpdate, RunProjectUpdate
+from awx.main.redact import UriCleaner
+from awx.main.constants import (
+    MINIMAL_EVENTS,
+)
+
+logger = logging.getLogger('awx.main.tasks.jobs')
+
+
+class RunnerCallback:
+    def event_handler(self, event_data):
+        #
+        # ⚠️  D-D-D-DANGER ZONE ⚠️
+        # This method is called once for *every event* emitted by Ansible
+        # Runner as a playbook runs.  That means that changes to the code in
+        # this method are _very_ likely to introduce performance regressions.
+        #
+        # Even if this function is made on average .05s slower, it can have
+        # devastating performance implications for playbooks that emit
+        # tens or hundreds of thousands of events.
+        #
+        # Proceed with caution!
+        #
+        """
+        Ansible runner puts a parent_uuid on each event, no matter what the type.
+        AWX only saves the parent_uuid if the event is for a Job.
+        """
+        # cache end_line locally for RunInventoryUpdate tasks
+        # which generate job events from two 'streams':
+        # ansible-inventory and the awx.main.commands.inventory_import
+        # logger
+        if isinstance(self, RunInventoryUpdate):
+            self.end_line = event_data['end_line']
```

Comment:
Right now there are variables such as ```end_line``` which belong to class ```BaseTask```. I am not sure if these variables are supposed to be moved to ```RunnerCallback``` class or left as is.

Reply:
https://github.com/ansible/awx/pull/11571/files#r793020460